### PR TITLE
Spreadsheet to cas refactoring

### PR DESCRIPTION
**Pull request by ubyndr**
Related with #34 

Implemented changes related with https://github.com/cellannotation/cas-tools/issues/34#issue-2110538012

### TODO
Validate schema compliance to ensure that the enhancement aligns with existing standards and expectations for dataset handling within our tools.

### CODE REVIEW

File: src/cas/spreadsheet_to_cas.py [hunk 1]
```diff
@@ -105,43 +155,73 @@ def spreadsheet2cas(
...
+
+    Note:
+        Requires `custom_lowercase_transform`, `get_cell_ids`, and column constants to be defined.
+    """
     stripped_data_result = raw_data_result.map(
         lambda x: x.strip() if isinstance(x, str) else x
     )
+    labelsets = OrderedDict()
     for index, row in stripped_data_result.iterrows():
         labelset = row[LABELSET_COLUMN]
```
Comment (hkir-dev) [inline src/cas/spreadsheet_to_cas.py:216]: According to spec "columns names MUST match CAS annotation object key names". We don't need these mappings in the code, they should match by name. 

Before the loop we can read the headers in the spreadsheet and match with the fields in the CAS schema. Then we can then use this matching while extracting data from the rows. We can update the test data to match column names with the schema fields.

File: src/cas/spreadsheet_to_cas.py [hunk 2]
```diff
@@ -152,21 +232,65 @@ def spreadsheet2cas(
             "cell_ontology_term": cell_label,
             "cell_ids": get_cell_ids(dataset_anndata, labelset, cell_label),
             "rationale": rationale,
-            "rationale_dois": meta_data_result["PAPER DOI"],
+            "rationale_dois": rationale_dois,
             "marker_gene_evidence": marker_gene_evidence,
```
Comment (hkir-dev) [inline src/cas/spreadsheet_to_cas.py:236]: can we use `|` separated list representation here.

File: src/cas/spreadsheet_to_cas.py [hunk 3]
```diff
@@ -34,18 +35,39 @@ def read_spreadsheet(file_path: str, sheet_name: Optional[str]):
...
+    raw_data = pd.DataFrame()
+
+    header_row_index = None
+    for index, row in spreadsheet_df.iterrows():
+        first_cell = str(row[0])
+        if first_cell.startswith("#"):
+            key = first_cell[1:].strip()
+            value = row[1] if pd.notnull(row[1]) else ""
+            if key in cap_schema["properties"]:
+                meta_data[key] = value
```
Comment (hkir-dev) [inline src/cas/spreadsheet_to_cas.py:57]: `author_name`, `orcid` and `author_list` is not populated. Also we should get type of the field from schema and parse accordingly. Value of `author_list` should be list not string.

**Review by hkir-dev — CHANGES_REQUESTED** [review]

File: src/test/spreadsheet_to_cas_test.py [hunk 4]
```diff
@@ -248,9 +248,9 @@ def test_spreadsheet2cas(self, mock_read_anndata_file, mock_download_source_h5ad
             with open(json_file_path, "r") as json_file:
                 json_data = json.load(json_file)
 
-            self.assertEqual(len(json_data), 10)
-            self.assertEqual(len(json_data["annotations"]), 103)
-            self.assertEqual(len(json_data["annotations"][0]), 14)
+            self.assertEqual(len(json_data), 4)
```
[the remark this answers is not in the document]
Comment (ubyndr) [reply src/test/spreadsheet_to_cas_test.py]: I've fixed this with my last [commit](https://github.com/cellannotation/cas-tools/pull/61/commits/730770483c8a378fd04246e861a4e30009df2696)

**Review by hkir-dev — APPROVED** [review]

**Review by mfutey — APPROVED** [review]

**Pull request merged**